File: Emolga/src/de/tectoast/emolga/commands/draft/DraftsetupCommand.java
package de.tectoast.emolga.commands.draft;

import de.tectoast.emolga.commands.Command;
import de.tectoast.emolga.commands.CommandCategory;
import de.tectoast.emolga.utils.draft.Draft;
import net.dv8tion.jda.api.entities.Member;
import net.dv8tion.jda.api.entities.Message;
import net.dv8tion.jda.api.entities.TextChannel;
import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent;

public class DraftsetupCommand extends Command {
    public DraftsetupCommand() {
        super("draftsetup", "`!draftsetup <Name> <Textchannel>` Startet das Draften der Liga in diesem Channel (im angegebenen Textchannel werden die Teams angezeigt)", CommandCategory.Draft);
    }

    @Override
    public void process(GuildMessageReceivedEvent e) {
        TextChannel tco = e.getChannel();
        Message m = e.getMessage();
        String msg = m.getContentDisplay();
        Member member = e.getMember();
        String name = msg.split(" ")[1];
        String tcid = m.getMentionedChannels().get(0).getId();
        new Draft(tco, name, tcid, false);
    }
}

File: src/main/kotlin/de/tectoast/emolga/features/draft/during/DraftsetupCommand.kt
package de.tectoast.emolga.features.draft.during

import de.tectoast.emolga.features.Arguments
import de.tectoast.emolga.features.CommandFeature
import de.tectoast.emolga.features.CommandSpec
import de.tectoast.emolga.features.InteractionData
import de.tectoast.emolga.league.League
import net.dv8tion.jda.api.Permission

object DraftsetupCommand : CommandFeature<DraftsetupCommand.Args>(
    ::Args,
    CommandSpec("draftsetup", "Startet das Draften der Liga in diesem Channel")
) {
    class Args : Arguments() {
        var name by string("Name", "Der Name der Liga")
        var switchdraft by boolean("switchdraft", "Ob es ein Switchdraft sein soll") {
            default = false
        }
        var nameguild by long("NameGuild", "Der Server, von dem die Namen geholt werden sollen").nullable()
    }

    init {
        restrict(flo)
    }

    context(iData: InteractionData)
    override suspend fun exec(e: Args) {
        League.executeOnFreshLock(e.name) {
            if (!iData.guild().selfMember.hasPermission(
                    iData.textChannel,
                    Permission.VIEW_CHANNEL,
                    Permission.MESSAGE_SEND
                )
            ) {
                return@executeOnFreshLock iData.reply(
                    "Ich habe keine Schreibrechte in diesem Channel!",
                    ephemeral = true
                )
            }
            startDraft(iData.textChannel, fromFile = false, switchDraft = e.switchdraft, nameGuildId = e.nameguild)
            iData.reply("+1", ephemeral = true)
        }
    }
}
